lib/partitions.js

```diff
@@ -130,6 +131,19 @@ Partition.prototype.produce = function() {
     }.bind(this));
 }
 
+/**
+ * Get a Partitions resource representing the partitions for this topic. This does not request any metadata since you may
+ * use the Partitions resource to get a specific partition; call list() on the resulting Partitions resource to get a
+ * full listing with metadata.
+ *
+ * @param res function(err,
+ */
+Partition.prototype.__defineGetter__("messages", function() {
+    if (!this._messages)
+        this._messages = new PartitionMessages(this);
+    return this._messages.messages.bind(this._messages);
```

Comment:
I agree that having `messages().messages({offset:0}, callback)` looks a bit odd. `messages().get({offset:0}, callback)` might not look as weird though. My example with `Partition` and `produce()` may not have been great -- a better example probably would have been something like `Partitions`'s `list()` and `Partition`'s `get()`, where we get the resource as an object, then the HTTP request is triggered via a method that is clearly associated with a specific HTTP verb.

As I said, I'm fine with this as is as long as we're convinced the `GET` is the only thing we would want. Otherwise, you want to leave open the possibility of implementing other methods for other HTTP verbs.